authorization/token/worker/token_cleanup_worker_blackbox_test.go

```diff
@@ -0,0 +1,59 @@
+package worker_test
+
+import (
+	"github.com/fabric8-services/fabric8-auth/authorization/token/worker"
+	"github.com/fabric8-services/fabric8-auth/gormtestsupport"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"testing"
+	"time"
+)
+
+type tokenCleanupWorkerBlackBoxTest struct {
+	gormtestsupport.DBTestSuite
+}
+
+func TestRunTokenCleanupWorkerBlackBoxTest(t *testing.T) {
+	suite.Run(t, &tokenCleanupWorkerBlackBoxTest{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+func (s *tokenCleanupWorkerBlackBoxTest) TestCleanupWorker() {
+	now := time.Now()
+	yesterday := now.AddDate(0, 0, -1)
+	tomorrow := now.AddDate(0, 0, 1)
+
+	// Clean up all tokens
+	require.NoError(s.T(), s.deleteAllTokens(s.T()))
+
+	s.Graph.CreateToken(yesterday)
+	s.Graph.CreateToken(yesterday)
+	s.Graph.CreateToken(tomorrow)
+
+	require.Equal(s.T(), 3, s.countTokens(s.T()))
+
+	// Start the worker with a 1 second timer
+	worker := worker.NewTokenCleanupWorker(s.Ctx, s.Application)
+
+	// Activate token cleanup once every hour
+	worker.Start(time.NewTicker(time.Second))
+	defer worker.Stop()
+
+	// Wait 5 seconds
+	time.Sleep(time.Duration(5) * time.Second)
```

Comment:
Don't waist 5 seconds on needless waiting in our test run time :) Start a timer with N milliseconds (10 for example) and check the results every 100 ms for up to 30 times or so in a loop. So the test execution time will be from 100ms up to 3s when it will fail by timeout.

Reply:
Good point, I've modified this to use a loop as suggested.  Worker timer I set to 50ms though, 10ms seems a little too aggressive ;)